fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoansApiResource.java

```diff
@@ -1294,4 +1350,20 @@ private String createLoanDelinquencyAction(Long loanId, ExternalId loanExternalI
         return delinquencyActionSerializer.serialize(result);
     }
 
+    private String modifyLoanApprovedAmount(Long loanId, ExternalId loanExternalId, String apiRequestBodyAsJson) {
+        Long resolvedLoanId = getResolvedLoanId(loanId, loanExternalId);
+        final CommandWrapperBuilder builder = new CommandWrapperBuilder().withJson(apiRequestBodyAsJson);
+        CommandWrapper commandRequest = builder.updateLoanApprovedAmount(resolvedLoanId).build();
+
+        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+        return this.toApiJsonSerializer.serialize(result);
+    }
+
+    private LoanApprovedAmountHistoryResponse getLoanApprovedAmountHistory(Long loanId, ExternalId loanExternalId) {
+        context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+        Long resolvedLoanId = getResolvedLoanId(loanId, loanExternalId);
+        List<LoanApprovedAmountHistoryData> loanApprovedAmountHistory = loanApprovedAmountHistoryRepository.findAllByLoanId(resolvedLoanId);
+        return new LoanApprovedAmountHistoryResponse(loanApprovedAmountHistory);
```

Comment:
We can return directly the `List<LoanApprovedAmountHistoryData>`, no need extra serialization.